core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentStatusManager.java

```diff
@@ -1039,17 +1039,19 @@ private static void writeLoadMetadata(AbsoluteTableIdentifier identifier,
     }
   }
 
-  private static ReturnTuple isUpdateRequired(boolean isForceDeletion, CarbonTable carbonTable,
-      AbsoluteTableIdentifier absoluteTableIdentifier, LoadMetadataDetails[] details) {
+  private static ReturnTuple isUpdateRequired(boolean cleanStaleInProgress, boolean
```

Comment:
@vikramahuja1001 : please update the document of clean files.

> 1 ) default clean files behaviour(clean files on table t1): clean MFD and Compacted segments will depend on query timeout(1 hr) and trashRetentionTimeout(7 days, default). 
2) clean files on table t1 options('force'='true'): clean MFD and Compacted segments immediately(Do not check for any timeout)
3) clean files on table t1 options('clean_inprgress'='true') : clean stale inprogress segments depends on trashRetentionTimeout, after 7 days(default behaviour)
4) clean files on table t1 options('clean_inprgress'='true', 'force'='true') : clean MFD, Compacted and stale inprogress segments immediately.(Do not check for any timeout)

**my comments for description** 
1) you have not mentioned about trash in any of this. mention what happens to trash in each. 
2)  And in your point 3) we should delete the expired MFD, compacted and expired trash folders also.

Reply:
1. Trash behaviour remains unchanged. Trash will be deleted automatically when option force = true, it's already in the document, If force = true is not given, then it will delete trash according to timestamp retention property.
2. in_progress = true, should only delete in_progress segments. Expired trash will be deleted anyways.

Reply:
If we don't clean expired MFD, expired compacted and expired trash along with expired stale_inprogress. we have to call clean files at least two times to clean all expired data. I feel it is not good. 

@QiangCai @akashrn5 : what you think ?

Reply:
@QiangCai @akashrn5 , can you clarify if we should delete the expired MFD, compacted in point 3?

Reply:
agree with @ajantha-bhat , the clean files with options, let it do its specific functionality, along with this, if others expired we should do that also, to save time for users.It should be like (1+2), (1+3) and (1+4). I think this will be better